### PR TITLE
Fix bug settled channel

### LINKED ISSUE #511: Could not identify the intended function with name `firstBlockAllowedToMonitor`
## Problem Definition
```
ms-goerli_1             | 2019-08-09 07:11:10.299418 [debug    ] Processing event               [monitoring_service.service] event_={'token_network_address': '0x070fEF2eA7cC30B91e5f0c650b78302B13a0E747', 'channel_identifier': 3, 'closing_participant': '0x7deA0108AB75a8CF373c02bdb237664A31fcf48B', 'block_number': 1036493, 'type_name': 'ReceiveChannelClosedEvent'}
ms-goerli_1             | Traceback (most recent call last):
ms-goerli_1             |   File "/usr/local/lib/python3.7/runpy.py", line 193, in _run_module_as_main
ms-goerli_1             |     "__main__", mod_spec)
ms-goerli_1             |   File "/usr/local/lib/python3.7/runpy.py", line 85, in _run_code
ms-goerli_1             |     exec(code, run_globals)
ms-goerli_1             |   File "/services/src/monitoring_service/cli.py", line 66, in <module>
ms-goerli_1             |     main(auto_envvar_prefix="MS")
ms-goerli_1             |   File "/usr/local/lib/python3.7/site-packages/click/core.py", line 764, in __call__
ms-goerli_1             |     return self.main(*args, **kwargs)
ms-goerli_1             |   File "/usr/local/lib/python3.7/site-packages/click/core.py", line 717, in main
ms-goerli_1             |     rv = self.invoke(ctx)
ms-goerli_1             |   File "/usr/local/lib/python3.7/site-packages/click/core.py", line 956, in invoke
ms-goerli_1             |     return ctx.invoke(self.callback, **ctx.params)
ms-goerli_1             |   File "/usr/local/lib/python3.7/site-packages/click/core.py", line 555, in invoke
ms-goerli_1             |     return callback(*args, **kwargs)
ms-goerli_1             |   File "/services/src/raiden_libs/cli.py", line 177, in call_with_blockchain_info
ms-goerli_1             |     return callback(**params)
ms-goerli_1             |   File "/services/src/raiden_libs/cli.py", line 122, in call_with_common_options_initialized
ms-goerli_1             |     return func(**params)
ms-goerli_1             |   File "/services/src/monitoring_service/cli.py", line 59, in main
ms-goerli_1             |     ms.start()
ms-goerli_1             |   File "/services/src/monitoring_service/service.py", line 126, in start
ms-goerli_1             |     self._process_new_blocks(last_block)
ms-goerli_1             |   File "/services/src/monitoring_service/service.py", line 160, in _process_new_blocks
ms-goerli_1             |     handle_event(event, self.context)
ms-goerli_1             |   File "/services/src/monitoring_service/service.py", line 56, in handle_event
ms-goerli_1             |     handler(event, context)
ms-goerli_1             |   File "/services/src/monitoring_service/handlers.py", line 131, in channel_closed_event_handler
ms-goerli_1             |     _first_allowed_block_to_monitor(event.token_network_address, channel, context) - 1
ms-goerli_1             |   File "/services/src/monitoring_service/handlers.py", line 96, in _first_allowed_block_to_monitor
ms-goerli_1             |     monitoring_service_address=context.ms_state.address,
ms-goerli_1             |   File "/usr/local/lib/python3.7/site-packages/web3/contract.py", line 1033, in __call__
ms-goerli_1             |     clone._set_function_info()
ms-goerli_1             |   File "/usr/local/lib/python3.7/site-packages/web3/contract.py", line 1042, in _set_function_info
ms-goerli_1             |     self.kwargs
ms-goerli_1             |   File "/usr/local/lib/python3.7/site-packages/web3/utils/contracts.py", line 128, in find_matching_fn_abi
ms-goerli_1             |     raise ValidationError(message)
ms-goerli_1             | web3.exceptions.ValidationError:
ms-goerli_1             | Could not identify the intended function with name `firstBlockAllowedToMonitor`, positional argument(s) of type `()` and keyword argument(s) of type `{'closed_at_block': <class 'int'>, 'settle_timeout': <class 'int'>, 'participant1': <class 'str'>, 'participant2': <class 'str'>, 'monitoring_service_address': <class 'bytes'>}`.
ms-goerli_1             | Found 1 function(s) with the name `firstBlockAllowedToMonitor`: ['firstBlockAllowedToMonitor(uint256,uint256,address,address,address)']
ms-goerli_1             | Function invocation failed due to no matching argument types.

### PR DESCRIPTION
**Pull request by palango**
The problem was that we changed the `last_known_block` in the `Context` from an independent value towards a a property returning the value of `latest_known_block` of `BlockchainState`, which is only updated when all events for a given block range are processed.

This led to the following condition to always be `False`.
```py
settle_period_end_block = event.block_number + channel.settle_timeout
    settle_period_over = settle_period_end_block < context.latest_confirmed_block
    if not settle_period_over:
       ....
```
This in turn let the MS to try to send an update to the blockchain for a closed channel.

We have tests for that behavior, however we wrongly updated the `BlockchainState` as well, which led to the tests not catching this.

This also includes some cleanups I did while fixing the bug.
Fixes #511

### CODE REVIEW

File: src/monitoring_service/handlers.py [hunk 1]
```diff
@@ -41,16 +41,23 @@
 class Context:
     ms_state: MonitoringServiceState
     db: Database
-    w3: Web3
+    web3: Web3
```
Comment (karlb) [inline src/monitoring_service/handlers.py]: This rename is something I also wanted to do for a while.

File: src/monitoring_service/handlers.py [hunk 2]
```diff
@@ -41,16 +41,23 @@
...
     monitoring_service_contract: Contract
     user_deposit_contract: Contract
     min_reward: int
     required_confirmations: int
 
     @property
-    def latest_known_block(self) -> BlockNumber:
+    def latest_commited_block(self) -> BlockNumber:
```
Comment (karlb) [inline src/monitoring_service/handlers.py]: Should we have both names `latest_commited_block` and `latest_known_block` for the same thing?

**Comment by karlb** [comment]
I relied to much an the tests when doing the change. Can you suggest a new test to catch this?

File: src/monitoring_service/handlers.py (hunk 2 above)
Comment (palango) [reply src/monitoring_service/handlers.py]: Good point, will change in `BlockchainState` as well

**Comment by palango** [comment]
> I relied to much an the tests when doing the change. Can you suggest a new test to catch this?

Yeah, so if I undo the change at https://github.com/raiden-network/raiden-services/pull/512/files#diff-6d7bdd065a1e0f2407c94540508e8707R123 then actually the test `test_channel_closed_event_handler_ignores_existing_channel_after_timeout` fails. It was unlucky that we changed the committed block at the same time.

That's why I chose not to add more tests, but we could do something like `test_first_allowed_monitoring` with a settled channel.

**Comment by karlb** [comment]
> Yeah, so if I undo the change at https://github.com/raiden-network/raiden-services/pull/512/files#diff-6d7bdd065a1e0f2407c94540508e8707R123 then actually the test `test_channel_closed_event_handler_ignores_existing_channel_after_timeout` fails. It was unlucky that we changed the committed block at the same time.

OK, that's a really unlucky case. Then the tests are probably fine as they are.

**Pull request merged**